### ISSUE TITLE
URLCache returns nothing before restart

### ISSUE THREAD

**Issue by denis-obukhov** [lorenzofiamingo/SwiftUI-CachedAsyncImage#6]
Thanks for the library. I try to use it in a scroll view with transaction animation when an image is loaded:

```
CachedAsyncImage(
    url: imageURL,
    urlCache: .imageCache,
    transaction: Transaction(animation: .default)
) { phase in
    switch phase {
    case .empty:
        ProgressView()
            .frame(height: 100)
    case .success(let image):
        image
              .resizable()
    case .failure:
        Image(systemName: "wifi.slash")
    @unknown default:
        EmptyView()
    }
}
.frame(maxHeight: 200)

private extension URLCache {
    static let imageCache = URLCache(
        memoryCapacity: 512 * 1000 * 1000,
        diskCapacity: 10 * 1000 * 1000 * 1000
    )
}

```

But once an image is loaded it continues to animate cell appearance (and other View content) when scrolling down and up again. URLCache doesn't return anything before the app relaunch.

**Comment by lorenzofiamingo** [comment]
Hi @denis-obukhov. Which version are you using?

**Comment by denis-obukhov** [comment]
@lorenzofiamingo v 1.10.1

**Comment by lorenzofiamingo** [comment]
@denis-obukhov I'm having problems trying to reproduce your error. 
Could you please modify my code in order to manifest it?

```swift
struct ContentView: View {
    
    let imageURL = URL(string: "https://picsum.photos/200/300")!
    
    var body: some View {
        ScrollView {
            CachedAsyncImage(
                url: imageURL,
                urlCache: .imageCache,
                transaction: Transaction(animation: .default)
            ) { phase in
                switch phase {
                case .empty:
                    ProgressView()
                        .frame(height: 500)
                case .success(let image):
                    image
                        .resizable()
                case .failure:
                    Image(systemName: "wifi.slash")
                @unknown default:
                    EmptyView()
                }
            }
            .frame(maxHeight: 500)
            Spacer(minLength: 9999)
        }
    }
}
```

**Comment by denis-obukhov** [comment]
@lorenzofiamingo Hm. Yes, it works ok in a project from the ground. I don't know why it works differently in my enterprise project. I'll add something if can reproduce it isolated

**Comment by denis-obukhov** [comment]
@lorenzofiamingo Here it is:

```
struct ContentView: View {
    let imageURL = URL(string: "https://idsb.tmgrup.com.tr/ly/uploads/images/2020/11/25/74707.jpg")!
    
    var body: some View {
        ScrollView(.vertical) {
            LazyVStack {
                ZStack {
                    Color.gray
                    
                    CachedAsyncImage(
                        url: imageURL,
                        urlCache: .imageCache,
                        transaction: Transaction(animation: .default)
                    ) { phase in
                        switch phase {
                        case .empty:
                            ProgressView()
                                .frame(height: 500)
                        case .success(let image):
                            image
                                .resizable()
                                .aspectRatio(contentMode: .fit)
                        case .failure:
                            Image(systemName: "wifi.slash")
                        @unknown default:
                            EmptyView()
                        }
                    }
                    .frame(maxHeight: 500)
                }
                
                ForEach(0..<1000) {
                    Text("Value: \($0)")
                        .foregroundColor(.white)
                        .background(Color.green)
                    
                }
            }
        }
    }
}
```

**Comment by lorenzofiamingo** [comment]
Still the problem is not manifesting...

The only problem I can imagine is that in the enterprise project the cache is filled or there is no space for it so it starts deleting some old images.

https://user-images.githubusercontent.com/40797951/153027461-a8440f00-4deb-4935-98b8-38b8d720ffdb.MP4

**Comment by lorenzofiamingo** [comment]
@denis-obukhov  Yes, I tried.
Where did you put `URLCache.imageCache.removeAllCachedResponses()`?
If I put it in the init of the App it behaves correctly.

**Comment by denis-obukhov** [comment]
@lorenzofiamingo Can you try this https://github.com/denis-obukhov/AsyncImageTest ?

**Comment by denis-obukhov** [comment]
I recorded a demo: https://youtu.be/RZ9bPZHx94c

**Comment by lorenzofiamingo** [comment]
Hi @denis-obukhov .
I saw the problem.

I think this is primarily a SwiftUI bug. `LazyVStack` for some reason is setting `phase` to its (empty) initial state (why the state is not preserved?* :/ ), and when it appears again the `init` is not called. Not only this behavior is fixed when the app relaunch, but also when the `ContentView` is presented, dismissed and presented again inside a `NavigationView`.

In the new version ([1.11](https://github.com/lorenzofiamingo/SwiftUI-CachedAsyncImage/tree/1.11)) I fixed this by checking the `resourceFetchType` of the `dataTask` metrics.
This is breaking animations when the `URL`\\`URLRequest` changes to a cached one, but I think it's not a common use-case to dynamically change the url.

Let me know if it works for you and if this makes sense.

*Maybe the `phase` is not preserved because SwiftUI is making some sort of optimization when it sees an `AsyncImagePhase` in the state of a `LazyVStack` children view.
